### PR TITLE
Updates Create Extension Service Wizard to generate service inspectors

### PR DESCRIPTION
**Pull request by Railboy**
Overview
---
Updates the wizard to generate an optional service inspector (#3773) for the new service.

ServiceInspectors

### CODE REVIEW

**Comment by StephenHodgson** [comment]
Awesome stuff as always @Railboy. I approve in spirit.

File: ExtensionTemplates/ExtensionInterfaceTemplate.txt [hunk 1]
```diff
@@ -3,8 +3,8 @@ using Microsoft.MixedReality.Toolkit.Extensions;
 
 namespace #NAMESPACE#
 {
-   public interface #INTERFACE_NAME# : IMixedRealityExtensionService
-    {
-        // Expose service features and abilities here
-    }
+	public interface #INTERFACE_NAME# : IMixedRealityExtensionService
```
Comment (keveleigh) [inline ExtensionTemplates/ExtensionInterfaceTemplate.txt]: Do these need to be tabs? Or can they be spaces, like the code formatting?

Comment (Railboy) [reply ExtensionTemplates/ExtensionInterfaceTemplate.txt]: I was getting some funky results with spaces. Not sure why.

Comment (wiwei) [reply ExtensionTemplates/ExtensionInterfaceTemplate.txt]: Any chance we could do a tabs -> spaces conversion in CreateTextAssetFromTemplate to fix that part up?

File: Assets/MixedRealityToolkit/Inspectors/ExtensionTemplates/ExtensionServiceCreator.cs [hunk 2]
```diff
@@ -401,6 +442,12 @@ public async Task BeginAssetCreationProcess()
             if (Result == CreateResult.Error)
                 return;
 
+            await Task.Delay(100);
```
Comment (wiwei) [inline Assets/MixedRealityToolkit/Inspectors/ExtensionTemplates/ExtensionServiceCreator.cs]: Some of the Task.Delays below make sense to me (i.e. waiting for compile) but these here I don't fully understand - CreateTextAssetFromTemplate appears to just be doing string replacement and WriteTextAssetToDisk LOOKS to be the sync flavor - Also it looks like each section is doing an unrelated thing (i.e. one for inspector, one for template, etc.)

What's the wait for (and could it be commented?). If it not necessary, can we remove it?

Usually anytime you see random 10-1000ms awaits in code it's a red flag.

File: ExtensionTemplates/ExtensionInterfaceTemplate.txt (hunk 1 above)
Comment (wiwei) [reply ExtensionTemplates/ExtensionInterfaceTemplate.txt]: Or I guess we just depend on peoples' text editors to convert the automagically.

File: Assets/MixedRealityToolkit/Inspectors/ExtensionTemplates/ExtensionServiceCreator.cs (hunk 2 above)
Comment (Railboy) [reply Assets/MixedRealityToolkit/Inspectors/ExtensionTemplates/ExtensionServiceCreator.cs]: Those waits serve no purpose other than to provide visual flow as assets are being created.

Comment (wiwei) [reply Assets/MixedRealityToolkit/Inspectors/ExtensionTemplates/ExtensionServiceCreator.cs]: Can you add a comment about this?

**Comment by wiwei** [comment]
I think I just want to get that minor comment in, and once that's done I'll handle pushing this into mrtk_development. Lemme know when I can handle that part.

**Comment by Railboy** [comment]
@wiwei Done.

**Pull request merged**